presto-redis/src/main/java/com/facebook/presto/redis/RedisSplitManager.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.redis;
+
+import com.facebook.presto.spi.ColumnHandle;
+import com.facebook.presto.spi.ConnectorPartition;
+import com.facebook.presto.spi.ConnectorPartitionResult;
+import com.facebook.presto.spi.ConnectorSplit;
+import com.facebook.presto.spi.ConnectorSplitManager;
+import com.facebook.presto.spi.ConnectorSplitSource;
+import com.facebook.presto.spi.ConnectorTableHandle;
+import com.facebook.presto.spi.FixedSplitSource;
+import com.facebook.presto.spi.HostAddress;
+import com.facebook.presto.spi.TupleDomain;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableList;
+import io.airlift.log.Logger;
+import redis.clients.jedis.Jedis;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+
+/**
+ * Redis specific implementation of {@link ConnectorSplitManager}.
+ */
+public class RedisSplitManager
+        implements ConnectorSplitManager
+{
+    private static final Logger log = Logger.get(RedisSplitManager.class);
+
+    private final String connectorId;
+    private final RedisConnectorConfig redisConnectorConfig;
+    private final RedisHandleResolver handleResolver;
+    private final RedisJedisManager jedisManager;
+
+    private static final long REDIS_MAX_SPLITS = 100;
+    private static final long REDIS_STRIDE_SPLITS = 100;
+
+    @Inject
+    public RedisSplitManager(@Named("connectorId") String connectorId,
+                             RedisConnectorConfig redisConnectorConfig,
+                             RedisHandleResolver handleResolver,
+                             RedisJedisManager jedisManager)
+    {
+        this.connectorId = checkNotNull(connectorId, "connectorId is null");
+        this.redisConnectorConfig = checkNotNull(redisConnectorConfig, "redisConfig is null");
+        this.handleResolver = checkNotNull(handleResolver, "handleResolver is null");
+        this.jedisManager = checkNotNull(jedisManager, "jedisManager is null");
+    }
+
+    @Override
+    public ConnectorPartitionResult getPartitions(ConnectorTableHandle tableHandle, TupleDomain<ColumnHandle> tupleDomain)
```

Comment:
The `ConnectorPartition` abstraction is deprecated (it's been replaced by TableLayout -- see TpchMetadata for an example of how it's used)